### PR TITLE
Remove `inline-merchandising`/`im` slot

### PR DESCRIPTION
**Pull request by deedeeh**
## What does this change?

This PR removes the `inline-merchandising` slot, aka `im`, and removes the "Inline Merchandising Slot" in Frontend tools.

It also removes the `hasInlineMerchandise` from the config so `guardian.config.page.hasInlineMerchandise` shouldn't exist anymore.

Brings in the commercial changes [v20.4.0](https://github.com/guardian/commercial/releases/tag/v20.4.0)

Other PRs related to this work https://github.com/guardian/commercial/pull/1516 and https://github.com/guardian/dotcom-rendering/pull/12104

## Checklist

- [x] Tested locally, and on CODE if necessary

### CODE REVIEW

**Review by deedeeh — COMMENTED** [review]
I will still need to bump Commercial with the changes, which I will do in this same PR once I have an approval for the Commercial PR and merge it.

**Review by arelra — APPROVED** [review]
✨

**Review by Jakeii — APPROVED** [review]

**Pull request merged**
